lib/ts/ingredients/emaildelivery/types.ts

```diff
@@ -14,25 +14,17 @@
  */
 
 export interface EmailService<T> {
-    sendEmail: (input: T, userConext: any) => Promise<void>;
+    sendEmail: (input: T & { userContext: any }) => Promise<void>;
 }
 
-export type RecipeInterface<T> = {
-    sendEmail: (input: T, userContext: any) => Promise<void>;
+export type IngredientInterface<T> = {
+    sendEmail: (input: T & { userContext: any }) => Promise<void>;
 };
 
-/**
- * input given by the user if emailDelivery config is passed in parent Recipe
- */
-export interface TypeConfigInput<T> {
-    service: EmailService<T>;
-    override?: (originalImplementation: RecipeInterface<T>) => RecipeInterface<T>;
-}
-
 /**
  * config class parameter when parent Recipe create a new EmailDeliveryRecipe object via constructor
  */
-export interface ConfigInput<T> {
+export interface TypeInput<T> {
     service: EmailService<T>;
-    recipeImpl: RecipeInterface<T>;
+    override?: (originalImplementation: IngredientInterface<T>) => IngredientInterface<T>;
```

Comment:
Do you prefer `RecipeInterface`? But later, when would be adding many other such ingredients (e.g. SMS). Don't you think it would be better not to call all of them as `RecipeInterface` but `IngredientInterface`?

Reply:
I agree. But i don't like `IngredientInterface` either. Let's discuss this on a call.